### PR TITLE
solver: added initial support for instance_deserialize in KPHP mode

### PR DESCRIPTION
**Pull request by i582**
At the moment, string assignment of the class
and assignment through the class constant are
supported:

```php
instance_deserialize ("", Foo::class);
instance_deserialize ("", "Foo");
```

### CODE REVIEW

File: src/tests/exprtype/exprtype_test.go [hunk 1]
```diff
@@ -2707,6 +2707,32 @@ function f1() {
 	runExprTypeTest(t, &exprTypeTestParams{code: code, stubs: "<?php /* no code */"})
 }
 
+func TestInstanceDeserializeType(t *testing.T) {
```
Comment (quasilyte) [inline src/tests/exprtype/exprtype_test.go:2710]: Consider to move this test to `kphp_test.go`.

Comment (i582) [reply src/tests/exprtype/exprtype_test.go]: Done

**Review by quasilyte — APPROVED** [review]
Looks legit.

File: src/solver/exprtype.go [hunk 2]
```diff
@@ -202,6 +202,41 @@ func binaryPlusOpType(sc *meta.Scope, cs *meta.ClassParseState, left, right ir.N
...
+		default:
+			return "", false
+		}
+	default:
+		return "", false
+	}
+
+	className, ok := GetClassName(cs, &ir.Name{
+		Value: name,
+	})
```
Comment (quasilyte) [inline src/solver/exprtype.go:232]: Can this literal be placed on the same line as GetClassName call?
```go
	className, ok := GetClassName(cs, &ir.Name{Value: name})
```

Comment (i582) [reply src/solver/exprtype.go]: Done

**Review by quasilyte — APPROVED** [review]

**Pull request merged**